pettingzoo/magent/magent_env.py

```diff
@@ -191,10 +188,12 @@ def state(self):
         return state
 
     def step(self, all_actions):
-        action_list = [0] * self.max_num_agents
+        action_list = [-1] * self.max_num_agents
         for i, agent in enumerate(self.possible_agents):
             if agent in all_actions:
                 action_list[i] = all_actions[agent]
+
+        action_list = list(filter(lambda v: v != -1, action_list))
```

Comment:
Instead of filtering, would it work if the agents were simply iterated over?

```
action_list = []
for i, agent in self.agents:
    action_list.append(all_actions[agent])
```

Reply:
Agree